### PR TITLE
AC-19, BriteVerify

### PR DESCRIPTION
**Pull request by knutsenm**

### CODE REVIEW

**Review by andrewroth — DISMISSED** [review]

File: app/lib/brite_verify.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "net/http"
+
+module BriteVerify
+  API_HOST = "bpi.briteverify.com"
+
+  def self.valid_email?(email_address)
+    Net::HTTP.start(API_HOST, 443, use_ssl: true) do |http|
+      uri = URI("https://#{API_HOST}/api/v1/fullverify")
+      req = Net::HTTP::Post.new(uri)
+      req["Authorization"] = "ApiKey: #{ENV["BRITE_VERIFY_API_KEY"]}"
```
[the remark this answers is not in the document]
Comment (knutsenm) [reply app/lib/brite_verify.rb:10]: No, I think you're right, we should force ourselves to have the key.

File: spec/lib/brite_verify_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      around do |example|
+        env_value_before = ENV["BRITE_VERIFY_API_KEY"]
+        ENV["BRITE_VERIFY_API_KEY"] = nil
+        example.run
+        ENV["BRITE_VERIFY_API_KEY"] = env_value_before
+      end
+
+      it "throws a KeyError" do
+        expect { subject }.to raise_error(KeyError)
```
Comment (knutsenm) [inline spec/lib/brite_verify_spec.rb:52]: The `raise_error` won't catch the error without the `{ }` :-/

Comment (andrewroth) [reply spec/lib/brite_verify_spec.rb:52]: It needs a block to raise something, I guess

**Review by andrewroth — DISMISSED** [review]

**Comment by knutsenm** [comment]
Tested well in staging. API secret is already set in production as well.

File: app/models/profile.rb [hunk 3]
```diff
@@ -36,8 +36,7 @@ def validate_required
 
   def validate_format
     if email_address # rubocop:disable Style/GuardClause
-      # Use same RegExp as Global Registry to catch invalid emails before sending there.
-      unless email_address.match?(/\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i)
+      unless BriteVerify.valid_email?(email_address)
         errors[email_address_name] = "Please enter a valid email address."
```
Comment (knutsenm) [inline app/models/profile.rb:40]: Matches the real-time message from the JS. The difference is this only appears after you hit Subscribe.

**Review by jbirdjavi — APPROVED** [review]

**Pull request merged**